tuf/api/metadata.py

```diff
@@ -0,0 +1,539 @@
+"""TUF role metadata model.
+
+This module provides container classes for TUF role metadata, including methods
+to read/serialize/write from and to JSON, perform TUF-compliant metadata
+updates, and create and verify signatures.
+
+TODO:
+ * Validation (some thoughts ...)
+   - Avoid schema, see secure-systems-lab/securesystemslib#183
+   - Provide methods to validate JSON representation (at user boundary)
+   - Fail on bad json metadata in read_from_json method
+   - Be lenient on bad/invalid metadata objects in memory, they might be
+     work in progress. E.g. it might be convenient to create empty metadata
+     and assign attributes later on.
+   - Fail on bad json metadata in write_to_json method, but with option to
+     disable check as there might be a justified reason to write WIP
+     metadata to json.
```

Comment:
Btw. what do others think about the [ValidationMixin](https://github.com/in-toto/in-toto/blob/74da7aabce663a5539ad3e1e17b2c871eeaa227a/in_toto/models/common.py#L27-L40) methods we use in in-toto? (also see [`Layout._validate_*`](https://github.com/in-toto/in-toto/blob/74da7aabce663a5539ad3e1e17b2c871eeaa227a/in_toto/models/layout.py#L420-L438) methods for example usage) 

I think it's a commendable way of doing this (credits to @SantiagoTorres)